### ISSUE TITLE
Disk section missing from HTTP /stats result in Riak 1.4.2

### ISSUE THREAD

**Issue by dmitrizagidulin** [basho/riak_kv#706]
The disk usage section is missing from the HTTP /stats api call in riak 1.4.2
For example, doing

```
curl http://localhost:8098/stats
```

results in:

``` javascript
{
   "riak_kv_stat_ts":1382377500,
   "vnode_gets":0,
   "vnode_gets_total":0,
   "vnode_puts":0,
   "vnode_puts_total":0,
   "vnode_index_reads":0,
   "vnode_index_reads_total":0,
   "vnode_index_writes":0,
   "vnode_index_writes_total":0,
   "vnode_index_writes_postings":0,
   "vnode_index_writes_postings_total":0,
   "vnode_index_deletes":0,
   "vnode_index_deletes_total":0,
   "vnode_index_deletes_postings":0,
   "vnode_index_deletes_postings_total":0,
   "node_gets":0,
   "node_gets_total":0,
   "node_get_fsm_siblings_mean":0,
   "node_get_fsm_siblings_median":0,
   "node_get_fsm_siblings_95":0,
   "node_get_fsm_siblings_99":0,
   "node_get_fsm_siblings_100":0,
   "node_get_fsm_objsize_mean":0,
   "node_get_fsm_objsize_median":0,
   "node_get_fsm_objsize_95":0,
   "node_get_fsm_objsize_99":0,
   "node_get_fsm_objsize_100":0,
   "node_get_fsm_time_mean":0,
   "node_get_fsm_time_median":0,
   "node_get_fsm_time_95":0,
   "node_get_fsm_time_99":0,
   "node_get_fsm_time_100":0,
   "node_puts":0,
   "node_puts_total":0,
   "node_put_fsm_time_mean":0,
   "node_put_fsm_time_median":0,
   "node_put_fsm_time_95":0,
   "node_put_fsm_time_99":0,
   "node_put_fsm_time_100":0,
   "read_repairs":0,
   "read_repairs_total":0,
   "coord_redirs_total":0,
   "executing_mappers":0,
   "precommit_fail":0,
   "postcommit_fail":0,
   "index_fsm_create":0,
   "index_fsm_create_error":0,
   "index_fsm_active":0,
   "list_fsm_create":0,
   "list_fsm_create_error":0,
   "list_fsm_active":0,
   "pbc_active":0,
   "pbc_connects":0,
   "pbc_connects_total":0,
   "node_get_fsm_active":0,
   "node_get_fsm_active_60s":0,
   "node_get_fsm_in_rate":0,
   "node_get_fsm_out_rate":0,
   "node_get_fsm_rejected":0,
   "node_get_fsm_rejected_60s":0,
   "node_get_fsm_rejected_total":0,
   "node_put_fsm_active":0,
   "node_put_fsm_active_60s":0,
   "node_put_fsm_in_rate":0,
   "node_put_fsm_out_rate":0,
   "node_put_fsm_rejected":0,
   "node_put_fsm_rejected_60s":0,
   "node_put_fsm_rejected_total":0,
   "leveldb_read_block_error":"undefined",
   "riak_pipe_stat_ts":1382377500,
   "pipeline_active":0,
   "pipeline_create_count":0,
   "pipeline_create_one":0,
   "pipeline_create_error_count":0,
   "pipeline_create_error_one":0,
   "cpu_nprocs":494,
   "cpu_avg1":225,
   "cpu_avg5":56,
   "cpu_avg15":28,
   "mem_total":16795246592,
   "mem_allocated":4261359616,
   "nodename":"riak@127.0.0.1",
   "connected_nodes":[

   ],
   "sys_driver_version":"2.0",
   "sys_global_heaps_size":0,
   "sys_heap_type":"private",
   "sys_logical_processors":16,
   "sys_otp_release":"R15B01",
   "sys_process_count":1530,
   "sys_smp_support":true,
   "sys_system_version":"Erlang R15B01 (erts-5.9.1) [source] [64-bit] [smp:16:16] [async-threads:64] [kernel-poll:true]",
   "sys_system_architecture":"x86_64-unknown-linux-gnu",
   "sys_threads_enabled":true,
   "sys_thread_pool_size":64,
   "sys_wordsize":8,
   "ring_members":[
      "riak@127.0.0.1"
   ],
   "ring_num_partitions":64,
   "ring_ownership":"[{'riak@127.0.0.1',64}]",
   "ring_creation_size":64,
   "storage_backend":"riak_kv_bitcask_backend",
   "erlydtl_version":"0.7.0",
   "riak_control_version":"1.4.2",
   "cluster_info_version":"1.2.4",
   "riak_search_version":"1.4.2",
   "merge_index_version":"1.3.1",
   "riak_kv_version":"1.4.2",
   "sidejob_version":"0.2.0",
   "riak_api_version":"1.4.2",
   "riak_pipe_version":"1.4.2",
   "riak_core_version":"1.4.2",
   "bitcask_version":"1.6.3",
   "basho_stats_version":"1.0.3",
   "webmachine_version":"1.10.4",
   "mochiweb_version":"1.5.1p6",
   "inets_version":"5.9",
   "erlang_js_version":"1.2.2",
   "runtime_tools_version":"1.8.8",
   "os_mon_version":"2.2.9",
   "riak_sysmon_version":"1.1.3",
   "ssl_version":"5.0.1",
   "public_key_version":"0.15",
   "crypto_version":"2.1",
   "sasl_version":"2.2.1",
   "lager_version":"2.0.0",
   "compiler_version":"4.8.1",
   "syntax_tools_version":"1.6.8",
   "stdlib_version":"1.18.1",
   "kernel_version":"2.15.1",
   "memory_total":61618136,
   "memory_processes":10265540,
   "memory_processes_used":10265526,
   "memory_system":51352596,
   "memory_atom":504409,
   "memory_atom_used":484167,
   "memory_binary":77352,
   "memory_code":10454345,
   "memory_ets":5616544,
   "riak_core_stat_ts":1382377501,
   "ignored_gossip_total":0,
   "rings_reconciled_total":0,
   "rings_reconciled":0,
   "gossip_received":0,
   "rejected_handoffs":0,
   "handoff_timeouts":0,
   "dropped_vnode_requests_total":0,
   "converge_delay_min":0,
   "converge_delay_max":0,
   "converge_delay_mean":0,
   "converge_delay_last":0,
   "rebalance_delay_min":0,
   "rebalance_delay_max":0,
   "rebalance_delay_mean":0,
   "rebalance_delay_last":0,
   "riak_kv_vnodes_running":64,
   "riak_kv_vnodeq_min":0,
   "riak_kv_vnodeq_median":0,
   "riak_kv_vnodeq_mean":0,
   "riak_kv_vnodeq_max":0,
   "riak_kv_vnodeq_total":0,
   "riak_pipe_vnodes_running":64,
   "riak_pipe_vnodeq_min":0,
   "riak_pipe_vnodeq_median":0,
   "riak_pipe_vnodeq_mean":0,
   "riak_pipe_vnodeq_max":0,
   "riak_pipe_vnodeq_total":0
}
```

Missing from this is the 'disk' section that was found in 1.4.1:

```
   "disk":[
      {
         "id":"/",
         "size":244277768,
         "used":86
      }
   ],
```

This is somewhat odd, in that, https://github.com/basho/riak_kv/blob/develop/src/riak_kv_wm_stats.erl#L87 looks like it's loading the disk stats.
And trying out the relevant code on the console in 1.4.2 yields:

```
root@riak1.4.2:~/dz/riak/rel/riak# bin/riak attach
Remote Shell: Use "Ctrl-C a" to quit. q() or init:stop() will terminate the riak node.
Erlang R15B01 (erts-5.9.1) [source] [64-bit] [smp:16:16] [async-threads:0] [kernel-poll:false]
Eshell V5.9.1  (abort with ^G)
(riak@127.0.0.1)1> {value, {disk, Disk}, Stats} = lists:keytake(disk, 1, riak_kv_stat:get_stats()).
{value,{disk,[{"/",191016620,7},
              {"/dev",8191988,1},
              {"/run",3280324,1},
              {"/run/lock",5120,1},
              {"/run/shm",8200804,0},
              {"/boot",92123,73}]},
       [{riak_kv_stat_ts,1382380565},
        {vnode_gets,0},
        {vnode_gets_total,0},
        {vnode_puts,0},
        {vnode_puts_total,0},
        {vnode_index_reads,0},
        {vnode_index_reads_total,0},
        {vnode_index_writes,0},
        {vnode_index_writes_total,0},
        {vnode_index_writes_postings,0},
        {vnode_index_writes_postings_total,0},
        {vnode_index_deletes,0},
        {vnode_index_deletes_total,0},
        {vnode_index_deletes_postings,0},
        {vnode_index_deletes_postings_total,0},
        {node_gets,0},
        {node_gets_total,0},
        {node_get_fsm_siblings_mean,0},
        {node_get_fsm_siblings_median,0},
        {node_get_fsm_siblings_95,0},
        {node_get_fsm_siblings_99,0},
        {node_get_fsm_siblings_100,0},
        {node_get_fsm_objsize_mean,0},
        {node_get_fsm_objsize_median,0},
        {node_get_fsm_objsize_95,...},
        {...}|...]}
(riak@127.0.0.1)2> Disk.
[{"/",191016620,7},
 {"/dev",8191988,1},
 {"/run",3280324,1},
 {"/run/lock",5120,1},
 {"/run/shm",8200804,0},
 {"/boot",92123,73}]
(riak@127.0.0.1)3> DiskFlat = [{struct, [{id, list_to_binary(Id)}, {size, Size}, {used, Used}]} || {Id, Size, Used} <- Disk].
[{struct,[{id,<<"/">>},{size,191016620},{used,7}]},
 {struct,[{id,<<"/dev">>},{size,8191988},{used,1}]},
 {struct,[{id,<<"/run">>},{size,3280324},{used,1}]},
 {struct,[{id,<<"/run/lock">>},{size,5120},{used,1}]},
 {struct,[{id,<<"/run/shm">>},{size,8200804},{used,0}]},
 {struct,[{id,<<"/boot">>},{size,92123},{used,73}]}]
(riak@127.0.0.1)6> rp(litss:pppend([Stats, [{disk, DiskFltt},, riak_core_stat:get_stats()])).
[{riak_kv_stat_ts,1382380565},
 {vnode_gets,0},
 {vnode_gets_total,0},
 {vnode_puts,0},
...
 {sys_system_version,<<"Erlang R15B01 (erts-5.9.1) [source] [64-bit] [smp:16:16] [async-threads:64] [kernel-poll:true]">>},
 {sys_system_architecture,<<"x86_64-unknown-linux-gnu">>},
 {sys_threads_enabled,true},
 {sys_thread_pool_size,64},
 {sys_wordsize,8},
 {ring_members,['riak@127.0.0.1']},
 {ring_num_partitions,64},
 {ring_ownership,<<"[{'riak@127.0.0.1',64}]">>},
 {ring_creation_size,64},
 {storage_backend,riak_kv_bitcask_backend},
 {erlydtl_version,<<"0.7.0">>},
 {riak_control_version,<<"1.4.2">>},
 {cluster_info_version,<<"1.2.4">>},
 {riak_search_version,<<"1.4.2">>},
 {merge_index_version,<<"1.3.1">>},
 {riak_kv_version,<<"1.4.2">>},
 {sidejob_version,<<"0.2.0">>},
 {riak_api_version,<<"1.4.2">>},
 {riak_pipe_version,<<"1.4.2">>},
 {riak_core_version,<<"1.4.2">>},
 {bitcask_version,<<"1.6.3">>},
 {basho_stats_version,<<"1.0.3">>},
 {webmachine_version,<<"1.10.4">>},
...
 {memory_binary,137640},
 {memory_code,10612983},
 {memory_ets,6004008},
 {disk,[{struct,[{id,<<"/">>},{size,191016620},{used,7}]},
        {struct,[{id,<<"/dev">>},{size,8191988},{used,1}]},
        {struct,[{id,<<"/run">>},{size,3280324},{used,1}]},
        {struct,[{id,<<"/run/lock">>},{size,5120},{used,1}]},
        {struct,[{id,<<"/run/shm">>},{size,8200804},{used,0}]},
        {struct,[{id,<<"/boot">>},{size,92123},{used,73}]}]},
 {riak_core_stat_ts,1382380754},
 {ignored_gossip_total,0},
 {rings_reconciled_total,0},
...
 {riak_pipe_vnodeq_total,0}]
ok
```

So, the disk section seems to be there in _that_ output..

**Comment by dmitrizagidulin** [comment]
Ok, at least I think I know the reason -- the 1.4.1 I was testing against was built from source. It looks like the PRs that fixed the 'disk' section in /stats (https://github.com/basho/riak_kv/commit/224fdab4182c904d5b28dafd6beba5f9a48febc7 / https://github.com/basho/riak_kv/pull/383 ) didn't get merged into the 1.4.1 / 1.4.2 release tags.

**Comment by dmitrizagidulin** [comment]
Sounds like the fix has been merged to go into Riak 2.0. Closing this issue..
